### PR TITLE
Rename quarkus-qute-server-pages to quarkus-qute-web

### PR DESCRIPTION
**Pull request by mkouba**
- related to [#35739](https://github.com/quarkusio/quarkus/issues/35739)
- rename the groupId from io.quarkiverse.quteserverpages to io.quarkiverse.qute.web
- rename the artifactId from quarkus-qute-server-pages to quarkus-qute-web
- rename the package from io.quarkiverse.quteserverpages to io.quarkiverse.qute.web
- rename the config prefix from quarkus.qsp to quarkus.qute-web
- rename class names with "Qsp" prefix to use the prefix "QuteWeb"
- convert config to ConfigMapping
- introduce new config property: web-templates-dir; defaulted to "web"
- changed the default value for root-path to "/"
- changed the default value for hidden-templates; no templates are hidden

### What's missing

1. Conditional dependencies - we need to add `dependencyCondition` to the core extensions.
2. I kept the `qsp:` namespace because I'm not sure if `q:` or `qw:` is a better alternative? But we could change it in a follow-up PR if needed.

### CODE REVIEW

**Review by ia3andy — COMMENTED** [review]
It looks good code wise, I will give it a try now

**Review by gastaldi — APPROVED** [review]

File: runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: QuteServerPages
+name: Qute Web
 description: Automatically serves qute templates via HTTP.
```
Comment (ia3andy) [inline runtime/src/main/resources/META-INF/quarkus-extension.yaml:2]:
```suggestion
description: Qute Web serves Qute templates over HTTP, easily pairing with RESTEasy (Reactive or Classic) for more precise control.
```

File: README.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Quarkus QuteServerPages
+# Quarkus Qute Web
 
 [![Version](https://img.shields.io/maven-central/v/io.quarkiverse.quteserverpages/quarkus-qute-server-pages.svg?label=Maven%20Central)](https://search.maven.org/artifact/io.quarkiverse.quteserverpages/quarkus-qute-server-pages)
 [![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
```
Comment (ia3andy) [inline README.md:4]: you need to change the doc here 👇

**Comment by ia3andy** [comment]
for `qsp:` namespace replacement, why not use `web:`?

File: README.md (hunk 2 above)
Comment (mkouba) [reply README.md:4]: I was waiting for the repo rename. We'll have to do some other stuff afterwards anyway...

**Comment by mkouba** [comment]
> for `qsp:` namespace replacement, why not use `web:`?

That's IMO too general :shrug: . And `qweb` is too long. Maybe just `q:` or `qu:`?

File: runtime/src/main/resources/META-INF/quarkus-extension.yaml (hunk 1 above)
Comment (mkouba) [reply runtime/src/main/resources/META-INF/quarkus-extension.yaml:2]: The _"easily pairing with RESTEasy (Reactive or Classic) for more precise control"_ part is not yet true so I was waiting for the follow-up PR to update the docs...

**Comment by ia3andy** [comment]
> > for `qsp:` namespace replacement, why not use `web:`?
> 
> That's IMO too general 🤷 . And `qweb` is too long. Maybe just `q:` or `qu:`?

Just looked at what's available `http:` seems like a very good candidate:
image

**Comment by mkouba** [comment]
> > > for `qsp:` namespace replacement, why not use `web:`?
> > 
> > 
> > That's IMO too general 🤷 . And `qweb` is too long. Maybe just `q:` or `qu:`?
> 
> Just looked at what's available `http:` seems like a very good candidate: image

That's actually a very good point. Let's use `http:` then... I'll update the PR shortly.

**Review by ia3andy — APPROVED** [review]
Ok added a few comments here and there. It works like a charm :)

**Pull request merged**